GCRCatalogs/agncombined.py

```diff
@@ -0,0 +1,47 @@
+"""
+AGN composite catalog (matched to cosmoDC2) reader
+
+This reader was designed by Yao-Yuan Mao and Eve Kovacs
+based a catalog provided by Scott Daniel.
+"""
+
+import re
+import os
+from GCRCatalogs.composite import CompositeReader
+import numpy as np
+
+def _calc_flux(mag):
+    return np.power(10, -0.4*mag)
+
+
+def get_composite_mag(original_mag, agn_mag):
+    if np.ma.isMaskedArray(agn_mag):
+        # fill masked array and convert to ndarray
+        agn_mag = agn_mag.filled(fill_value=np.inf)
+    # add agn flux to original flux
+    return -2.5*np.log10(_calc_flux(original_mag) + _calc_flux(agn_mag))
+
+
+class AGNCombinedCatalog(CompositeReader):
+
+    def _subclass_init(self, **kwargs):
+
+        """
+        AGNComposite catalog reader, inherited from CompositeCatalog class 
+        """
+        self._catalog_names = [cat.identifier for cat in self._catalogs]
+
+        for band in 'ugriz':
+             self._quantity_modifiers['mag_{}_lsst'.format(band)] = (
+                 get_composite_mag,
+                 (self._catalog_names[0], 'mag_{}_lsst'.format(band)),
+                 (self._catalog_names[1], 'mag_{}_lsst(agn)'.format(band)),
+                 )
+             self._quantity_modifiers['mag_{}_noagn_lsst'.format(band)] = (
+                 self._catalog_names[0], 'mag_{}_lsst'.format(band)
+             )
+        
+        suppress_overwrite = kwargs.get('suppress_overwrite', None) #schema variables from main catalog
+        if suppress_overwrite:
+            for q in suppress_overwrite:
+                self._quantity_modifiers[q] = (self._catalog_names[0], q)
```

Comment:
to avoid unintended overwrite if `suppress_overwrite` is not correctly specified 

```suggestion
                if self._quantity_modifiers.get(q) == (self._catalog_names[1], q):
                    self._quantity_modifiers[q] = (self._catalog_names[0], q)
```

Reply:
Hmm, there is no guarantee that the variable name delivered by the reader for the second catalog will match the native name of that quantity. Take for example 'redshift'. The reader for the second catalog has 'redshift' as a non-native and a native quantity.  But the native quantity could have been called 'z' instead, and I would still want to do the override, because  'redshift' is a quantity name in the second catalog that matches a quantity name in the first catalog.  Wouldn't the above test fail for this example, leading to no suppression of the overwrite?

Reply:
Not sure if I follow. In the second catalog, regardless `redshift` is native or not, it will overwrite the first catalog's `redshift` (also regardless native or not). In this case, in the composite catalog, `redshift` will point to the second catalog (i.e., `(self._catalog_names[1], 'redshift')`).

Reply:
Sorry, not sure I understand. I thought that native quantities in the composite catalog are always tuples and hence guaranteed to be unique. The only possibility of a name collision is if the reader for catalog1 defines a quantity with the same name as the reader for catalog 2.  If there is a name collision, default behavior is to point to the catalog 2? Do I have that correct?

Reply:
Yes, that's correct. Hence the code I suggested first check if the quantity is already pointing to the quantity in the second catalog (i.e., `if self._quantity_modifiers.get(q) == (self._catalog_names[1], q)`), and if so, then set it to the first catalog instead (`self._quantity_modifiers[q] = (self._catalog_names[0], q)`).

Reply:
OK, so suppose the native quantity in the second catalog is called 'z',  but the reader for the second catalog renames it to 'redshift'. This collides with 'redshift' for the first catalog and we would like to prevent the overwrite. But if we use the test, doesn't it fail?  (because self._quantity_modifiers.get('redshift') == (self._catalog_names[1], 'z') and not (self._catalog_names[1], 'redshift') ).  Sorry if I have misunderstood, but I don't think we can assume that the native name and the reader name are always the same.

Reply:
In the scenario that you describe, both `redshift` and `z` will appear in the composite catalog, and `redshift` will point to `(self._catalog_names[1], 'redshift')` while `z` will point to `(self._catalog_names[1], 'z')`

Reply:
Ah, thank you I hadn't realized that both of the names would appear. I thought that 'z' would only appear in the tuple. In that case, I am fine with adding the check. I will modify the code with all the suggested changes and do a final test when nersc comes back up.